### PR TITLE
IH-74 Update api valid user ticket to use coupon

### PR DESCRIPTION
**Pull request by Sotatek-KimMai**

### CODE REVIEW

File: src/main/java/art/heredium/service/TicketPayService.java [hunk 1]
```diff
@@ -77,14 +85,62 @@ public void handleError(ClientHttpResponse response) {}
...
     return PaymentsValidResponse.from(entity);
   }
 
+  private void applyCouponDiscount(Ticket ticket, String couponUuid, Long accountId) {
+    CouponUsage couponUsage =
+        couponUsageRepository
+            .findByUuid(couponUuid)
+            .orElseThrow(() -> new ApiException(ErrorCode.COUPON_NOT_FOUND));
+
+    if (couponUsage.getIsUsed()) {
```
Comment (Sotatek-ThoLe) [inline src/main/java/art/heredium/service/TicketPayService.java:109]: Nếu là isPermanent thì vẫn cho dùng tiếp.

File: src/main/java/art/heredium/domain/ticket/entity/Ticket.java [hunk 2]
```diff
@@ -48,6 +49,7 @@
 
 @Entity
 @Getter
+@Setter
```
Comment (Sotatek-ThoLe) [inline src/main/java/art/heredium/domain/ticket/entity/Ticket.java:52]: Bỏ @Setter, tạo public method riêng:
```java
public void setPrice(long price){this.price = price};

File: src/main/java/art/heredium/service/TicketPayService.java (hunk 1 above)
Comment (Sotatek-KimMai) [reply src/main/java/art/heredium/service/TicketPayService.java]: I updated in https://github.com/Incuvers-IT/heredium-BE/pull/51/commits/3f019e338052f38d5ac6177d1fe28f7e0b3b9971

File: src/main/java/art/heredium/domain/ticket/entity/Ticket.java (hunk 2 above)
Comment (Sotatek-KimMai) [reply src/main/java/art/heredium/domain/ticket/entity/Ticket.java]: I updated in https://github.com/Incuvers-IT/heredium-BE/pull/51/commits/f6cf6d92be434a3a803eef7d2edd67996d8ca45e

**Review by Sotatek-ThoLe — APPROVED** [review]

**Pull request merged**